### PR TITLE
Ta1 zan

### PR DESCRIPTION
**Pull request by KelllyChen**
video link: https://prodduke-my.sharepoint.com/:v:/r/personal/tj152_duke_edu/Documents/Attachments/ta1_spotify_api%201.mp4?csf=1&web=1&e=RfJZBS&nav=eyJyZWZlcnJhbEluZm8iOnsicmVmZXJyYWxBcHAiOiJTdHJlYW1XZWJBcHAiLCJyZWZlcnJhbFZpZXciOiJTaGFyZURpYWxvZy1MaW5rIiwicmVmZXJyYWxBcHBQbGF0Zm9ybSI6IldlYiIsInJlZmVycmFsTW9kZSI6InZpZXcifX0%3D

requirement:
certifi==2024.8.30
charset-normalizer==3.3.2
contourpy==1.3.0
cycler==0.12.1
fonttools==4.53.1
idna==3.8
kiwisolver==1.4.5
matplotlib==3.9.2
numpy==2.1.0
packaging==24.1
pillow==10.4.0
pyparsing==3.1.4
python-dateutil==2.9.0.post0
redis==5.0.8
requests==2.32.3
six==1.16.0
spotipy==2.24.0
urllib3==2.2.2

**Features**
- Pull data from playlists, artists, albums, and tracks
- Search for an artist by name.
- Fetch and display the artist's details including followers and genres.
- Retrieve all albums and tracks of the artist.
- Display the popularity score of each track and calculate average popularity score of each album.
- Identify and display the top 5 most popular tracks of the artist.
- Plot a graph showing the average popularity of each album by release year.

**Get Spotify API Credentials:**

Register your application on the [Spotify Developer Dashboard](https://developer.spotify.com/dashboard/applications) to obtain your client_id and client_secret.
Update Credentials:

**Replace the placeholders in the script with your Spotify API client_id and client_secret:**

client_id = 'YOUR_CLIENT_ID'
client_secret = 'YOUR_CLIENT_SECRET'

**Input:**
When prompted, enter the name of the artist you want to analyze.

**Output:**

- The script will display the artist's details(including number of followers and genres), each album name, release dates, popularity score of each song, and average popularity score of each album.
- It will also print the top 5 most popular songs.
- A plot will be shown, visualizing the average popularity of each album by release year.

### CODE REVIEW

**Comment by tejasj02** [comment]
Get Spotify API Credentials:

Register your application on the [Spotify Developer Dashboard](https://developer.spotify.com/dashboard/applications) to obtain your client_id and client_secret.
Update Credentials:

Replace the placeholders in the script with your Spotify API client_id and client_secret:

client_id = 'YOUR_CLIENT_ID'
client_secret = 'YOUR_CLIENT_SECRET'

File: api_spotify.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
+#Code was constructed with Spotipy API and ChatGPT referencing SpotipyAPI.
+
+import spotipy
+from spotipy.oauth2 import SpotifyClientCredentials
+import matplotlib.pyplot as plt
+import heapq
+
+# Replace with your client ID and client secret
+client_id = 'f32e47f002124474834dc28b01450218'
+client_secret = 'bd65693f357544e794e1090acccdfc46'
```
Comment (dayeon3117) [inline api_spotify.py]: It would probably be a good idea to remove the client ids and secrets from your script.

**Comment by brinnaebent** [comment]
@KelllyChen please add me to the sharing of the video, I am unable to access.

**Comment by tejasj02** [comment]
@brinnaebent I apologize about that, the link should work for you now.

**Comment by brinnaebent** [comment]
Nice job on the video demo, very clear 15/15

**Review by medinardaniel — COMMENTED** [review]
Loose code (-5). 25/30

**Pull request merged**